src/server/injected/injectedScript.ts

```diff
@@ -319,21 +319,42 @@ export class InjectedScript {
     });
   }
 
+  private _isFillableElement(element: Element) {
+    return element.nodeName.toLowerCase() === 'input' ||
+      element.nodeName.toLowerCase() === 'textarea' ||
+      (element as HTMLElement).isContentEditable;
+  }
+
   waitForEnabledAndFill(node: Node, value: string): InjectedScriptPoll<FatalDOMError | 'error:notconnected' | 'needsinput' | 'done'> {
+    let lastTarget: Element | undefined;
+
     return this.pollRaf((progress, continuePolling) => {
       if (node.nodeType !== Node.ELEMENT_NODE)
         return 'error:notelement';
 
-      if (node && node.nodeName.toLowerCase() !== 'input' &&
-          node.nodeName.toLowerCase() !== 'textarea' &&
-          !(node as any).isContentEditable) {
+      let element: Element | undefined = node as Element;
+      if (!this._isFillableElement(element)) {
         // Go up to the label that might be connected to the input/textarea.
-        node = (node as Element).closest('label') || node;
+        element = element.closest('label') || element;
+      }
+      element = this.findLabelTarget(element);
+      if (element && !this._isFillableElement(element)) {
+        let child = element;
+        // Go down a single child route to find the input/textarea.
+        while (child.childElementCount === 1)
```

Comment:
```suggestion
        while (child.childElementCount === 1 && !this._isFillableElement(child))
```
Otherwise this will exhibit strange behavior of contentEditable elements with one element child. For example `<div contentEditable=true><br></div>`